src/custom_gates.jl

```diff
@@ -1,3 +1,15 @@
+@doc raw"""
+    DoubleExcitation(ϕ)
+
+Generate the matrix representation of the [DoubleExcitation](https://docs.pennylane.ai/en/stable/code/api/pennylane.DoubleExcitation.html) gate.
+
+This gate performs an SO(2) rotation in the subspace {|1100⟩, |0011⟩}, transforming the states as follows:
+
+```math
+|0011\rangle & \rightarrow \cos\left(\frac{\phi}{2}\right)|0011\rangle + \sin\left(\frac{\phi}{2}\right)|1100\rangle \\
+|1100\rangle & \rightarrow \cos\left(\frac{\phi}{2}\right)|1100\rangle - \sin\left(\frac{\phi}{2}\right)|0011\rangle
+```
+"""
```

Comment:
Would be good to add a doctest here (and in other docstrings)

Reply:
Thanks. I will add doctests to verify the equations, Example:

```
julia> m * [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0]
16-element StaticArraysCore.SVector{16, ComplexF64} with indices SOneTo(16):
                 0.0 + 0.0im
                 0.0 + 0.0im
                 0.0 + 0.0im
 -0.9781966068080447 + 0.0im
                 0.0 + 0.0im
                 0.0 + 0.0im
                 0.0 + 0.0im
                 0.0 + 0.0im
                 0.0 + 0.0im
                 0.0 + 0.0im
                 0.0 + 0.0im
                 0.0 + 0.0im
 -0.2076810016087838 + 0.0im
                 0.0 + 0.0im
                 0.0 + 0.0im
                 0.0 + 0.0im

julia> m * [0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]
16-element StaticArraysCore.SVector{16, ComplexF64} with indices SOneTo(16):
                 0.0 + 0.0im
                 0.0 + 0.0im
                 0.0 + 0.0im
 -0.2076810016087838 + 0.0im
                 0.0 + 0.0im
                 0.0 + 0.0im
                 0.0 + 0.0im
                 0.0 + 0.0im
                 0.0 + 0.0im
                 0.0 + 0.0im
                 0.0 + 0.0im
                 0.0 + 0.0im
  0.9781966068080447 + 0.0im
                 0.0 + 0.0im
                 0.0 + 0.0im
                 0.0 + 0.0im
```

Reply:
Sample DocTest:

```
julia> ϕ = 3.56;
julia> gate_matrix = DoubleExcitation(3.56);
julia> m  = matrix_rep(gate_matrix);
julia> eq1 = m * [0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]
julia> eq2 = m * [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0]
julia> eq1 ==  [0, 0, 0, cos(3.56/2), 0, 0, 0, 0, 0, 0, 0, 0, sin(3.56/2), 0, 0, 0]
true
julia> eq2 ==  [0, 0, 0, -sin(3.56/2), 0, 0, 0, 0, 0, 0, 0, 0, cos(3.56/2), 0, 0, 0]
true
```